File: kinotic-js/workspace/packages/persistence/src/api/decorators/EntityDecorator.ts
import { EntityType } from '@/api/decorators/EntityType'
import { C3Decorator } from '@kinotic-ai/idl'
import { MultiTenancyType } from '@/api/decorators/MultiTenancyType'

/**
 * Signifies that a class is an entity.
 */
export class EntityDecorator extends C3Decorator {
    public multiTenancyType: MultiTenancyType = MultiTenancyType.NONE
    public entityType: EntityType = EntityType.TABLE
    constructor() {
        super()
        this.type = 'Entity'
    }

    public withMultiTenancyType(type: MultiTenancyType): EntityDecorator {
        this.multiTenancyType = type
        return this
    }

    public withEntityType(type: EntityType): EntityDecorator {
        this.entityType = type
        return this
    }
}

File: kinotic-persistence/src/main/java/org/kinotic/persistence/api/model/idl/decorators/EntityDecorator.java
package org.kinotic.persistence.api.model.idl.decorators;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.kinotic.idl.api.schema.decorators.C3Decorator;
import org.kinotic.idl.api.schema.decorators.DecoratorTarget;

import java.util.List;

/**
 * Created by Navíd Mitchell 🤪on 6/16/23.
 */
@Getter
@Setter
@Accessors(chain = true)
public final class EntityDecorator extends C3Decorator {

    @JsonIgnore
    public static final String type = "Entity";

    private MultiTenancyType multiTenancyType = MultiTenancyType.NONE;

    private EntityType entityType = EntityType.TABLE;

    public EntityDecorator() {
        this.targets = List.of(DecoratorTarget.TYPE);
    }
}
